### ISSUE TITLE
Public tables are never deleted

### ISSUE THREAD

**Issue by mauriciovasquezbernal** [iovisor/bcc#1171]
I was looking at the code that handles the table creation and destruction, I cannot find any place where public tables are destroyed. In fact the code at [1] works, note that after the first program has been destroyed, the second one is still able to use the public table.

Is that behavior excepted?
@drzaeus77 

Thanks,
Mauricio

[1] https://gist.github.com/mauriciovasquezbernal/844ceca4a3ba48941c1a28fffc161dc5

**Comment by drzaeus77** [comment]
No, I would consider this a bug, but one that I was aware of when pushing the initial implementation.
